### ISSUE TITLE
panic: StoreBlockHeight (5200791) > StateBlockHeight + 1 (1)

### ISSUE THREAD

**Issue by alexmeise** [cosmos/gaia#1719]
## Summary of Bug

Hello, I cant seem to be able to start gaiad by following this instructions:

install guide: https://hub.cosmos.network/main/getting-started/installation.html
download quicksync snap https://hub.cosmos.network/main/getting-started/quickstart.html
join mainnet: https://hub.cosmos.network/main/hub-tutorials/join-mainnet.html

Getting the following ERROR: 
.....
```
panic: StoreBlockHeight (5200791) > StateBlockHeight + 1 (1)
```
.....

full log: 
```

rk@instance-2:~/.gaia/config$ gaiad start --x-crisis-skip-assert-invariants
11:12AM INF starting node with ABCI Tendermint in-process
11:12AM INF Starting multiAppConn service impl=multiAppConn module=proxy
11:12AM INF Starting localClient service connection=query impl=localClient module=abci-client
11:12AM INF Starting localClient service connection=snapshot impl=localClient module=abci-client
11:12AM INF Starting localClient service connection=mempool impl=localClient module=abci-client
11:12AM INF Starting localClient service connection=consensus impl=localClient module=abci-client
11:12AM INF Starting EventBus service impl=EventBus module=events
11:12AM INF Starting PubSub service impl=PubSub module=pubsub
11:12AM INF Starting IndexerService service impl=IndexerService module=txindex
11:12AM INF ABCI Handshake App Info hash= height=0 module=consensus protocol-version=0 software-version=v7.0.2
11:12AM INF ABCI Replay Blocks appHeight=0 module=consensus stateHeight=0 storeHeight=5200791
11:12AM INF created new capability module=ibc name=ports/transfer
11:12AM INF port binded module=x/ibc/port port=transfer
11:12AM INF claimed capability capability=1 module=transfer name=ports/transfer
panic: StoreBlockHeight (5200791) > StateBlockHeight + 1 (1)

goroutine 1 [running]:
github.com/tendermint/tendermint/consensus.(*Handshaker).ReplayBlocks(_, {{{0xb, 0x0}, {0xc000beff80, 0x8}}, {0xc000beff90, 0xb}, 0x4f5b97, 0x0, {{0x0, ...}, ...}, ...}, ...)
	github.com/tendermint/tendermint@v0.34.19/consensus/replay.go:382 +0xad9
github.com/tendermint/tendermint/consensus.(*Handshaker).Handshake(0xc015833fd0, {0x25224e0, 0xc000162680})
	github.com/tendermint/tendermint@v0.34.19/consensus/replay.go:268 +0x3c8
github.com/tendermint/tendermint/node.doHandshake({_, _}, {{{0xb, 0x0}, {0xc000beff80, 0x8}}, {0xc000beff90, 0xb}, 0x4f5b97, 0x0, ...}, ...)
	github.com/tendermint/tendermint@v0.34.19/node/node.go:325 +0x1b8
github.com/tendermint/tendermint/node.NewNode(0xc0010ae140, {0x2512750, 0xc0004935e0}, 0xc000f688a0, {0x24fdc80, 0xc0010d99c8}, 0x0?, 0x0?, 0xc000f68c00, {0x2516658, ...}, ...)
	github.com/tendermint/tendermint@v0.34.19/node/node.go:733 +0x577
github.com/cosmos/cosmos-sdk/server.startInProcess(_, {{0x0, 0x0, 0x0}, {0x2528f48, 0xc00106d9e0}, {0x0, 0x0}, {0x2519768, 0xc000f69c20}, ...}, ...)
	github.com/cosmos/cosmos-sdk@v0.45.4/server/start.go:272 +0x7db
github.com/cosmos/cosmos-sdk/server.StartCmd.func2(0xc000fdfb80?, {0xc0010768a0?, 0x0?, 0x1?})
	github.com/cosmos/cosmos-sdk@v0.45.4/server/start.go:126 +0x169
github.com/spf13/cobra.(*Command).execute(0xc000fdfb80, {0xc001076890, 0x1, 0x1})
	github.com/spf13/cobra@v1.4.0/command.go:856 +0x67c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000f1ac80)
	github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.4.0/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	github.com/spf13/cobra@v1.4.0/command.go:895
github.com/cosmos/cosmos-sdk/server/cmd.Execute(0x0?, {0xc000f67280, 0xe})
	github.com/cosmos/cosmos-sdk@v0.45.4/server/cmd/execute.go:36 +0x1eb
main.main()
	github.com/cosmos/gaia/v7/cmd/gaiad/main.go:16 +0x2c
```

## Version

`git clone -b v7.0.2 https://github.com/cosmos/gaia.git`

```
rk@instance-2:~$ uname -a
Linux instance-2 5.15.0-1016-gcp #21~20.04.1-Ubuntu SMP Fri Aug 5 12:53:07 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
rk@instance-2:~$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.4 LTS"
```

## Steps to Reproduce

in a clean ubuntu 20.04 machine first install: 

```
sudo apt-get update
sudo apt-get install -y make gcc
curl -OL https://golang.org/dl/go1.18.2.linux-amd64.tar.gz
sudo tar -C /usr/local -xvf go1.18.2.linux-amd64.tar.gz
export PATH=$PATH:/usr/local/go/bin
export PATH=$PATH:$(go env GOPATH)/bin
git clone -b v7.0.2 https://github.com/cosmos/gaia.git
cd gaia && make install
gaiad version

```

then download quicksync snap via:  https://hub.cosmos.network/main/getting-started/quickstart.html

```
mkdir $HOME/.gaia/config -p
sudo apt-get install wget liblz4-tool aria2 jq -y
export URL=`curl -L https://quicksync.io/cosmos.json|jq -r '.[] |select(.file=="cosmoshub-4-pruned")|.url'`
echo $URL
cd $HOME/.gaia/data
aria2c -x5 $URL
lz4 -c -d basename $URL | tar xf -
wget https://quicksync.io/addrbook.cosmos.json
mv addrbook.cosmos.json $HOME/.gaia/config/addrbook.json
export GENESIS_URL="https://github.com/cosmos/mainnet/blob/master/genesis/genesis.cosmoshub-4.json.gz?raw=true"
wget -O genesis.cosmoshub-4.json.gz $GENESIS_URL
gzip -d genesis.cosmoshub-4.json.gz
mv genesis.cosmoshub-4.json ~/.gaia/config/genesis.json
gaiad start --x-crisis-skip-assert-invariants

```

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

**Comment by alexmeise** [comment]
Tried with updated ubuntu: 

```
rk@cosmos-test:~$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"
```


and received: 
```

1:25PM INF minted coins from module account amount=4309334uatom from=mint module=x/bank
panic: UnmarshalJSON cannot decode empty bytes

goroutine 82 [running]:
github.com/cosmos/cosmos-sdk/x/params/types.Subspace.Get({{0x25204f8, 0xc000062fc0}, 0xc000010a88, {0x2502850, 0xc0010c4480}, {0x25028a0, 0xc0010c4540}, {0xc001060480, 0x9, 0x1f}, ...}, ...)
```

On the next run I received again the same error: 

```
1:48PM INF claimed capability capability=1 module=transfer name=ports/transfer
panic: StoreBlockHeight (5200791) > StateBlockHeight + 1 (1)

goroutine 1 [running]:
github.com/tendermint/tendermint/consensus.(*Handshaker).ReplayBlocks(_, {{{0xb, 0x0}, {0xc000d92850, 0x8}}, {0xc000d92860, 0xb}, 0x4f5b97, 0x0, {{0x0, ...}, ...}, ...}, ...)
	github.com/tendermint/tendermint@v0.34.19/consensus/replay.go:382 +0xad9

```

**Comment by okwme** [comment]
hi @alexmeise thanks for the issue.
sorry it's taken a minute to find the time to try reproducing the error.
i've noticed that your commands have some fixes that were missing in our docs, i appreciate the excuse to update our docs.
while you've got some corrections i think you've also got a mistake in one part that was also missing in ours.
basically the commands around downloading the different resources from quicksync are not explicit enough in where those resources should go.
when following your instructions verbatim the data directory is incorrectly put at `$HOME/.gaia/data/data` instead of `$HOME/.gaia/data`. When I put the data into the correct spot i was able to successfully sync with the network.

your other errors listed from your other attempts at joining the network all look like they have to do with having corrupted data directories that were partially or incorrectly synced with the network.

I've made [this PR](https://github.com/cosmos/gaia/pull/1757) to outline where the differences are, hopefully that clarifies where your path went wrong and helps get you online.

**Comment by Andiami-Yusaka** [comment]
I faced the same error.

I reviewed previously asked questions and tested the below solutions, but none of them worked:

- Changed the gaiad version to 4, 7, 10, 12, 14. Only version 7 and 4 work and raise the above error. Other binaries face a segmentation error.

- Followed all steps in the below script except initialization. I had synchronized around 12M blocks using the gaiad 7.
`https://github.com/cosmos/gaia/blob/main/contrib/statesync.bash`

- Used unsafe-reset-all parameter to reset the node database

I run this command:
`/usr/local/bin/gaiad-v4.1.0-linux-amd64 start --x-crisis-skip-assert-invariants --home /mnt/md0/production/nodes/cosmos/data/`

The data and config folders are in the above path.
